client/implementation/src/main/java/io/smallrye/graphql/client/impl/typesafe/json/JsonReader.java

```diff
@@ -43,7 +44,12 @@ Object read() {
             return readErrorOr();
         if (isListOfErrors(value) && !isGraphQlErrorsType())
             throw cantApplyErrors(readGraphQlClientErrors());
-        return reader(location).read();
+        Reader<?> reader = reader(location);
+        if (reader == null) {
+            throw new InvalidResponseException(
+                    "invalid " + type.getTypeName() + " value for " + location.getDescription() + ": " + value);
+        }
+        return reader.read();
```

Comment:
Wouldn't it be easier to throw this exception directly in line 88 instead of returning `null`?

Reply:
Line 103 returns `null` when it can't match any reader, so I took that as "the contract" of this method and stuck to it. Of course we can change it.

Reply:
I think we should throw two exceptions. Returning `null` in line 103 would have caused a NPE before. A helpful exception would be better there, and it's different from the case in line 88.

Reply:
You're welcome to fix it ;)